app/views/contexts/section_summary_context.py

```diff
@@ -122,10 +122,26 @@ def _custom_summary_elements(self, section_summary, current_location, section):
 
     def _list_summary_element(self, summary, current_location, section) -> Mapping:
         current_list = self._list_store[summary["for_list"]]
-        list_collector_block = self._schema.get_list_collector_for_list(
-            section, for_list=summary["for_list"]
+
+        routing_path = self._router.routing_path(
+            section["id"], current_location.list_item_id
         )
 
+        list_collector_blocks_on_path = [
+            list_collector_block
+            for list_collector_block in self._schema.get_list_collectors_for_list(
+                section, for_list=summary["for_list"]
+            )
+            if list_collector_block["id"] in routing_path.block_ids
+        ]
+
+        if not list_collector_blocks_on_path:
+            list_collector_block = self._schema.get_list_collector_for_list(
+                section, for_list=summary["for_list"]
+            )
+        else:
+            list_collector_block = list_collector_blocks_on_path[0]
```

Comment:
This evaluates both sides, so I don't see a reason to check the negative side first. Also, this is potentially fetching the list twice when nothing is on the path. But ultimately, if none of the list collectors are on the path then calling 
```py
			self._schema.get_list_collector_for_list(
                section, for_list=summary["for_list"]
            )
```
won't help as that is already not on the path. If none of the collectors are on the path, then you should fetch the driving question.

Reply:
Refactored it so it's not negative first, waiting for James and his review before I make any more changes.